osquery/tables/networking/curl_certificate.cpp

```diff
@@ -127,9 +221,51 @@ static void fillRow(Row& r, X509* cert) {
 
     r["sha1_fingerprint"] = ss.str();
   }
+
+  r["certificate_version"] = INTEGER(certversion(cert));
+  r["signature_algorithm"] = signature_algorithm(cert);
+  r["signature"] = signature(cert);
+
+  // get the authority and subject key identifier
+  // It will be same for self-signed certificate
+  r["subject_key_identifier"] =
+      certificate_extensions(cert, NID_subject_key_identifier);
+  r["authority_key_identifier"] =
+      certificate_extensions(cert, NID_authority_key_identifier);
+
+  r["key_usage"] = certificate_extensions(cert, NID_key_usage);
+  r["extended_key_usage"] = certificate_extensions(cert, NID_ext_key_usage);
+  r["certificate_policies"] =
+      certificate_extensions(cert, NID_certificate_policies);
+
+  r["subject_alternative_names"] =
+      certificate_extensions(cert, NID_subject_alt_name);
+  r["issuer_alternative_names"] =
+      certificate_extensions(cert, NID_issuer_alt_name);
+
+  r["info_access"] = certificate_extensions(cert, NID_info_access);
+  r["subject_info_access"] = certificate_extensions(cert, NID_sinfo_access);
+  r["policy_mappings"] = certificate_extensions(cert, NID_policy_mappings);
+
+  r["has_expired"] = INTEGER(has_cert_expired(cert));
+
+  r["basic_constraint"] = certificate_extensions(cert, NID_basic_constraints);
+  r["name_constraints"] = certificate_extensions(cert, NID_name_constraints);
+  r["policy_constraints"] =
+      certificate_extensions(cert, NID_policy_constraints);
+
+  // set the dump_certificate flag
+  r["dump_certificate"] = INTEGER(dump_certificate);
+
+  // check the dump_certificate flag and dump the certificate in PEM format
+  if (dump_certificate) {
+    r["certificate_pem"] = pem(cert);
+  }
 }
 
-Status getTLSCertificate(std::string hostname, QueryData& results) {
+Status getTLSCertificate(std::string hostname,
```

Comment:
It looks like this `hostname` should be passed by const reference `const std::string& hostname`. Do you mind making this change while you're improving the table in this PR?